pkg/runtime/kubernetes/environment_create.go

```diff
@@ -64,72 +67,184 @@ func (p generalProvisioner) EnvironmentCreate(ctx context.Context,
 	hostKeyPath := "/var/envd/hostkey"
 	authKeyPath := "/var/envd/authkey"
 	var defaultPermMode int32 = 0666
-	expectedPod := v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        env.Name,
-			Namespace:   p.namespace,
-			Labels:      labels,
-			Annotations: annotations,
+
+	codeDirectoryVolumeMount := v1.VolumeMount{
+		Name:      "code-dir",
+		MountPath: fmt.Sprintf("/home/envd/%s", projectName),
+	}
+
+	containers := []v1.Container{
+		{
+			Name:  "envd",
+			Image: env.Spec.Image,
+			Ports: []v1.ContainerPort{
+				{
+					Name:          "ssh",
+					ContainerPort: 2222,
+				},
+			},
+			Env: []v1.EnvVar{
+				{
+					Name:  "ENVD_HOST_KEY",
+					Value: hostKeyPath,
+				},
+				{
+					Name:  "ENVD_AUTHORIZED_KEYS_PATH",
+					Value: authKeyPath,
+				},
+				{
+					Name:  "ENVD_WORKDIR",
+					Value: fmt.Sprintf("/home/envd/%s", projectName),
+				},
+			},
+			VolumeMounts: []v1.VolumeMount{
+				{
+					Name:      "secret",
+					ReadOnly:  true,
+					MountPath: hostKeyPath,
+					SubPath:   "hostkey",
+				},
+				{
+					Name:      "secret",
+					ReadOnly:  true,
+					MountPath: authKeyPath,
+					SubPath:   "publickey",
+				},
+				codeDirectoryVolumeMount,
+			},
+			Resources: v1.ResourceRequirements{
+				Requests: resRequest,
+				Limits:   resRequest,
+			},
 		},
-		Spec: v1.PodSpec{
-			Containers: []v1.Container{
+	}
+
+	volumes := []v1.Volume{
+		{
+			Name: "secret",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName:  "envd-server",
+					DefaultMode: &defaultPermMode,
+				},
+			},
+		},
+		{
+			Name: "code-dir",
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{},
+				// HostPath: &v1.HostPathVolumeSource{
```

Comment:
Could we remove the comments?

Reply:
The comments were not added for this pull request, but i can remove them.